src/librustc_middle/ty/query/on_disk_cache.rs

```diff
@@ -374,6 +401,35 @@ impl<'sess> OnDiskCache<'sess> {
         debug_assert!(prev.is_none());
     }
 
+    pub fn get_raw_def_id(&self, hash: &DefPathHash) -> Option<RawDefId> {
+        self.foreign_def_path_hashes.get(hash).copied()
+    }
+
+    pub fn try_remap_cnum(&self, tcx: TyCtxt<'_>, cnum: u32) -> Option<CrateNum> {
```

Comment:
Perhaps some part of `def_path_hash_to_def_id` can be moved to `on_disk_cache.rs` to avoid exposing all this stuff (`fn get_raw_def_id`, `fn try_remap_cnum`) to other modules.

Reply:
I ended up moving all of this to `on_disk_cache.rs`